examples/nemo/scripts/utils.py

```diff
@@ -287,93 +287,43 @@ def insert_explicit_loop_parallelism(
                         for ref in lp.stop_expr.walk(Reference))
                  or (str(len(loop.walk(Loop))) !=
                      loop.stop_expr.arguments[1].value))):
-            print("ICE Loop not parallelised for performance reasons")
+            loop.append_preceding_comment(
+                "PSyclone: ICE Loop not parallelised for performance reasons")
             continue
+
         # Skip if looping over ice categories, ice or snow layers
         # as these have only 5, 4, and 1 iterations, respectively
         if (any(ref.symbol.name in ('jpl', 'nlay_i', 'nlay_s')
                 for ref in loop.stop_expr.walk(Reference))):
-            print("Loop not parallelised because stops at 'jpl', 'nlay_i' "
-                  "or 'nlay_s'.")
+            loop.append_preceding_comment(
+                "PSyclone: Loop not parallelised because stops at 'jpl',"
+                " 'nlay_i' or 'nlay_s'.")
             continue
 
-        def skip_for_correctness(loop):
-            for call in loop.walk(Call):
-                if not isinstance(call, IntrinsicCall):
-                    print(f"Loop not parallelised because it has a call to "
-                          f"{call.routine.name}")
-                    return True
-                if not call.is_available_on_device():
-                    print(f"Loop not parallelised because it has a "
-                          f"{call.intrinsic.name} not available on GPUs.")
-                    return True
-            if loop.walk(CodeBlock):
-                print("Loop not parallelised because it has a CodeBlock")
-                return True
-            return False
-
         # If we see one such ice linearised loop, we assume
         # calls/codeblocks are not a problem (they are not)
-        if not any(ref.symbol.name in ('npti',)
-                   for ref in loop.stop_expr.walk(Reference)):
-            if skip_for_correctness(loop):
-                continue
+        # if any(ref.symbol.name in ('npti',)
+        #            for ref in loop.stop_expr.walk(Reference)):
+        #     opts = {"force": True}
 
         # pnd_lev requires manual privatisation of ztmp
```

Comment:
This is not new but presumably just ignoring the deps on 'ztmp' doesn't fix the need for manual privatisation?

Reply:
Removing this did not decrease the performance*.

* It could have been done for a large number of OpenMP threads in the CPU for the ECMWF case. But I prefer the better reliability now that we target more versions and models.

I also created #2671 for it, but @arporter points out there is other solutions.